src/antsibull_build/changelog.py

```diff
@@ -763,6 +825,63 @@ def _populate_ansible_changelog(
                 )
 
 
+def _cleanup_collection_version(
+    collection_name: str,
+    collection_data: dict[str, RemoveCollectionVersionSchema],
```

Comment:
`_cleanup_collection_changelogs` passes a dict with `Semver` keys, not `str` keys.

```suggestion
    collection_data: dict[SemVer, RemoveCollectionVersionSchema],
```

Reply:
I'm quite surprised that the type checker didn't notice this.

Reply:
Hmm, for some reason mypy things that there's `Any` instead of `SemVer`:
```
src/antsibull_build/changelog.py:866: note: Revealed type is "Union[builtins.dict[builtins.str, builtins.dict[Any, antsibull_build.changelog.RemoveCollectionVersionSchema]], None]"
```
`reveal_type(SemVer)` yields `Any` as well. Maybe it's because semantic_version has no type hints? (https://github.com/rbarrois/python-semanticversion/issues/138)